pkg/registration/register/csr/hub_driver.go

```diff
@@ -190,30 +197,40 @@ func NewCSRHubDriver(
 		}
 
 		if !v1CSRSupported && v1beta1CSRSupported {
-			csrApprover.controller = NewCSRApprovingController[*certificatesv1beta1.CertificateSigningRequest](
+			csrDriverForHub.controller = NewCSRApprovingController[*certificatesv1beta1.CertificateSigningRequest](
 				kubeInformers.Certificates().V1beta1().CertificateSigningRequests().Informer(),
 				kubeInformers.Certificates().V1beta1().CertificateSigningRequests().Lister(),
 				newCSRV1beta1Approver(kubeClient),
 				csrReconciles,
 				recorder,
 			)
 			recorder.Eventf("CSRV1beta1Approver", "Using v1beta1 CSR api to manage managed cluster client certificate")
-			return csrApprover, nil
+			return csrDriverForHub, nil
 		}
 	}
 
-	csrApprover.controller = NewCSRApprovingController[*certificatesv1.CertificateSigningRequest](
+	csrDriverForHub.controller = NewCSRApprovingController[*certificatesv1.CertificateSigningRequest](
 		kubeInformers.Certificates().V1().CertificateSigningRequests().Informer(),
 		kubeInformers.Certificates().V1().CertificateSigningRequests().Lister(),
 		newCSRV1Approver(kubeClient),
 		csrReconciles,
 		recorder,
 	)
 
-	return csrApprover, nil
+	return csrDriverForHub, nil
 }
 
 func (a *CSRHubDriver) CreatePermissions(_ context.Context, _ *clusterv1.ManagedCluster) error {
 	// noop
 	return nil
 }
+
+func (c *CSRHubDriver) Accept(cluster *clusterv1.ManagedCluster) bool {
+	return c.allows(cluster)
+}
+
+func (c *CSRHubDriver) allows(cluster *clusterv1.ManagedCluster) bool {
+	_, isManagedClusterArnPresent := cluster.Annotations["agent.open-cluster-management.io/managed-cluster-arn"]
```

Comment:
csr hub driver should not consider these annotations. I think what this needs to do is to check the related csr on the hub, and if there is existing one which is created by the approved user, the cluster can be accepted. 

cc @skeeey

Reply:
Existing acceptance of csr ManagedCluster doesn't check anything related to csr. It would be a new feature.

Reply:
I think the problem here is we get managedcluster object in managedcluster controller but not csr, but this controller is triggered based on changes to o ly managedcluster cr.

We will.have to make another api call to fetch csr. Would that be acceptable?

Also there can be multiple csrs for a single spoke but only one Managedcluster cr, how will we decide that which csr to look at?

Seems like there is additional complication involved here, request other registration experts to help implement here.

Reply:
We added this annotation condition here to check that anything that doesn't fir awsirsa is a csr request.

Reply:
I think for `CSRHubDriver. Accept` we should return `true` directly  and don‘t do other checks to keep the behavior: https://github.com/open-cluster-management-io/ocm/blob/main/pkg/registration/hub/managedcluster/controller.go#L140

Reply:
> I think for CSRHubDriver. Accept we should return true

I think the same. I think the aggreatedAccept func should check all driver accepts it, it should make an AND rather than OR check. And in isra driver, it should return true if the cluster does not have certain annotations

Reply:
@qiujian16 do you mean, it should return **false**(not true) if the cluster does not have certain annotations? I believe that is the only way ANDing them would work instead of ORing.

Reply:
Your suggestion works, please disregard above message.